modules/src/erc721/erc721_base.rs

```diff
@@ -0,0 +1,142 @@
+use crate::erc721::errors::Error;
+use crate::erc721::events::{Approval, ApprovalForAll, Transfer};
+use crate::erc721::Erc721;
+use odra::contract_env::{caller, revert};
+use odra::types::event::OdraEvent;
+use odra::types::{Address, U256};
+use odra::{Mapping, UnwrapOrRevert};
+
+#[odra::module]
+pub struct Erc721Base {
+    // Erc721 base fields.
+    pub balances: Mapping<Address, U256>,
+    pub owners: Mapping<U256, Option<Address>>,
+    pub token_approvals: Mapping<U256, Option<Address>>,
+    pub operator_approvals: Mapping<(Address, Address), bool>
+}
+
+impl Erc721 for Erc721Base {
+    fn balance_of(&self, owner: Address) -> U256 {
+        if owner.is_zero() {
```

Comment:
We don't need to mimic solidity zero address. We use an `Option` instead.

Reply:
Yes, but still ERC721 states that a zero address is illegal and shouldn't receive any tokens.

Reply:
Yes, but still, in none of the current and future backends, the idea of zero address is nonexistent. We shouldn't be that orthodox and just read the balance. Frankly, each platform creates its own standard, based on erc's but not so strict.